presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkRowBatch.java

```diff
@@ -204,8 +333,9 @@ private RowTupleSupplier(int partitionCount, int rowCount, byte[] rowData, int[]
             this.totalSize = totalSize;
 
             this.rowData = ByteBuffer.wrap(requireNonNull(rowData, "rowData is null"));
+            this.rowData.order(LITTLE_ENDIAN);
```

Comment:
hmm? what's this for? :)

Reply:
To make sure we are writing `short` in `LITTLE_ENDIAN` byte order, as the default byte order is `BIG_ENDIAN` that is different that the machine byte order.